src/darker/__main__.py

```diff
@@ -56,45 +60,70 @@ def format_edited_parts(
-            # 9. A re-formatted Python file which produces an identical AST was
-            #    created successfully - write an updated file or print the diff if
-            #    there were any changes to the original
-            content_after_reformatting = _reformat_single_file(
+    with concurrent.futures.ThreadPoolExecutor(max_workers=jobs or None) as executor:
+        futures: List[
+            concurrent.futures.Future[ProcessedDocument]
+        ] = []  # pylint: disable=unsubscriptable-object
+        for path_in_repo in sorted(changed_files):
+            enable_black = (git_root / path_in_repo) in files_to_blacken
+            future = executor.submit(
+                _format_single_file,
                 git_root,
                 path_in_repo,
                 revrange,
-                rev2_content,
-                rev2_isorted,
+                black_config if enable_black else None,
                 enable_isort,
-                black_config,
             )
-        else:
-            # File was excluded by Black configuration, don't reformat
-            content_after_reformatting = rev2_isorted
-        if report_unmodified or content_after_reformatting != rev2_content:
-            yield (src, rev2_content, content_after_reformatting)
+            futures.append(future)
+
+        for future in concurrent.futures.as_completed(futures):
+            src, rev2_content, content_after_reformatting = future.result()
+            if report_unmodified or content_after_reformatting != rev2_content:
+                yield (src, rev2_content, content_after_reformatting)
+
+
+def _format_single_file(
+    git_root: Path,
+    path_in_repo: Path,
+    revrange: RevisionRange,
+    black_config: BlackConfig,
+    enable_isort: bool,
+) -> ProcessedDocument:
+    src = git_root / path_in_repo
+    rev2_content = git_get_content_at_revision(path_in_repo, revrange.rev2, git_root)
+    # 1. run isort
+    if enable_isort:
+        rev2_isorted = apply_isort(
+            rev2_content,
+            src,
+            black_config.get("config"),
+            black_config.get("line_length"),
+        )
+    else:
+        rev2_isorted = rev2_content
+    if black_config is not None:
+        # 9. A re-formatted Python file which produces an identical AST was
+        #    created successfully - write an updated file or print the diff if
+        #    there were any changes to the original
+        content_after_reformatting = _reformat_single_file(
+            git_root,
+            path_in_repo,
+            revrange,
+            rev2_content,
+            rev2_isorted,
+            enable_isort,
+            black_config,
+        )
+    else:
+        # File was excluded by Black configuration, don't reformat
+        content_after_reformatting = rev2_isorted
+    return src, rev2_content, content_after_reformatting
 
 
 def _reformat_single_file(  # pylint: disable=too-many-arguments,too-many-locals
```

Comment:
It would make sense to rename this function now to `_apply_black()` since `_format_single_file()` was now introduced and we also already have `apply_isort()`.